docs/key-issues/gpus.md

```diff
@@ -180,3 +204,87 @@ Potential solutions on the PyPI side include:
 - make an environment marker or selector package approach work,
 - improve interoperability with other package managers, in order to be able to
   declare a dependency on a CUDA or ROCm version as externally provided,
+
+
+## Additional Notes on CUDA Compatibility
+
+Compatibility across CUDA versions is a common problem with numerous pitfalls to be aware of.
+There are three primary components of CUDA:
+
+1. The CUDA Toolkit (CTK): This component includes the CUDA runtime library
+   (libcudart.so) along with a range of other libraries and tools including
+   math libraries like cuBLAS. libcudart.so and a few other core headers and
+   libraries compose the bare minimum required to compile CUDA code.
+2. The user-mode driver: This is the libcuda.so library. This library is
+   required to actually run CUDA code.
+3. The kernel-mode driver: The nvidia.ko file. This constitutes what is
+   typically considered a "driver" in common parlance when referring to other
+   peripherals connected to a computer.
+
+CUDA developers and users usually do not need to be aware of the kernel-mode driver.
+For the rest of this section, when referring to the "driver" we will always be referring to the user-mode driver libcuda.so.
+
+The CUDA runtime library makes no forward or backward compatibility
+guarantees, meaning that libraries that dynamically link to the CUDA runtime
+may not work correctly if they are run on a system with a different CUDA
+runtime shared library than the one they were compiled against. In this scenario,
+users are responsible for having the right CUDA runtime library installed.  As
+a result, the official CUDA recommendation is to statically link the CUDA
+runtime (see
+[here](https://docs.nvidia.com/cuda/cuda-c-best-practices-guide/index.html#recommendations-for-building-a-minor-version-compatible-library)
+and
+[here](https://docs.nvidia.com/cuda/cuda-c-best-practices-guide/index.html#distributing-the-cuda-runtime-and-libraries)
+for more information).
+
+CUDA drivers have always been backward compatible. Any code that runs with some
+driver version X is installed will always run correctly with some newer driver
+version Y>X. As briefly discussed above, as of CUDA 11.0 CUDA also promises
+[minor version
+compatibility](https://docs.nvidia.com/deploy/cuda-compatibility/index.html#minor-version-compatibility)
+(MVC). This compatibility guarantees that CUDA code compiled using a certain
+version of the CTK will be compatible with any driver version within the same
+major release. This behavior is useful because it is often easier for users to
+upgrade their CUDA runtime than it is to upgrade the driver, especially on
+shared machines. For instance, the CTK may be installed using conda, while the
+driver library cannot be. An example of leveraging MVC would be compiling code
```

Comment:
This is changing nowadays. NVidia is [working](https://github.com/conda-forge/staged-recipes/issues/21382) to redistribute its entire[^1] setup through conda-forge.

[^1]: AFAIU; I admit that I don't know if the actual driver library will be part of `cuda`, but it would be consistent with distributing pretty much everything else that used to not be redistributable.

Reply:
Yup, both Leo and I are deeply involved in that effort (FYI we are both NVIDIA employees). The driver library will unfortunately still not be part of the conda-forge offering, but the rest of the CTK will be on cf.